src/ElasticSearch.php

```diff
@@ -121,26 +111,26 @@ public function query($queryString, $type, $start = 0, $count = 10, $order = '_i
 
         try {
             $query = $this->getQueryBuilder()->getSearchParams($queryString, $type, $start, $count, $order, $dir);
-            $this->getLogger()->debug('Query ', $query);
+            $this->getLogger()->debug(sprintf('Elasticsearch Query %s', json_encode($query)));
 
             return $this->buildResultSet(
                 $this->getClient()->search(
                     $query
                 )
             );
-        } catch (\Exception $exception) {
+        } catch (Exception $exception) {
             switch ($exception->getCode()) {
                 case 400:
                     $json = json_decode($exception->getMessage(), true);
                     $message = __(
                         'There is an error in your search query, system returned: %s',
-                        $json['error']['reason']
+                        $json['error']['reason'] ?? ''
                     );
-                    $this->getLogger()->error($message, [$exception->getMessage()]);
+                    $this->getLogger()->error(sprintf('Elasticsearch: %s %s', $message, $exception->getMessage()));
```

Comment:
Do you think it helps to put in the logs the exception FQN too? Sometimes the message is too generic and we do not know from where it comes :)